frame/support/procedural/src/construct_runtime/expand/executive.rs

```diff
@@ -0,0 +1,107 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License
+
+use crate::construct_runtime::parse::ExecutiveSection;
+use frame_support_procedural_tools::generate_crate_access_2018;
+use proc_macro2::{Ident, TokenStream};
+use quote::quote;
+use syn::Result;
+
+pub fn expand_executive(
+	runtime: &Ident,
+	system: &Ident,
+	scrate: &TokenStream,
+	block: &TokenStream,
+	executive_section: ExecutiveSection,
+) -> Result<TokenStream> {
+	let executive = generate_crate_access_2018("frame-executive")?;
+
+	let migrations = match executive_section.migration_section {
+		Some(migrations) => quote!(#migrations),
+		None => quote!(()),
+	};
+
+	let try_runtime_section = if let Ok(try_runtime) =
+		generate_crate_access_2018("frame-try-runtime")
+	{
+		quote! {
+			#[cfg(feature = "try-runtime")]
+			impl #runtime {
+				fn api_impl_try_runtime_upgrade(checks: #try_runtime::UpgradeCheckSelect) -> Result<Weight, #scrate::sp_runtime::TryRuntimeError> {
+					Executive::try_runtime_upgrade(checks)
+				}
+
+				fn api_impl_try_execute_block(
+					block: #block,
+					state_root_check: bool,
+					signature_check: bool,
+					select: #try_runtime::TryStateSelect
+				) -> Result<Weight, &'static str> {
+					Executive::try_execute_block(block, state_root_check, signature_check, select)
+				}
+			}
+		}
+	} else {
+		quote!()
+	};
+
+	let res = quote! {
+		/// Executive: handles dispatch to the various modules.
+		pub type Executive = #executive::Executive<
+			#runtime,
+			#block,
+			#system::ChainContext<#runtime>,
```

Comment:
This is kinda cheating, if you want to be pedantic. There's now no good way to configure this. But I am not sure if this is all that useful.

Reply:
Yeah, I will add the support to provide a custom Executive as well, using the syntax you suggested above.